### PR TITLE
Peer Review

### PR DESCRIPTION
**Pull request by Jchinonso**
#### What does this PR do?
- Peer Review
#### Description of Task to be completed?
- All recent changes

### CODE REVIEW

File: server/controllers/GroupCtr.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          }
+        });
+      } else {
+        return res.status(400).send({
+          message: 'Group does not exist'
+        });
+      }
+    });
+  },
+
```
Comment (DanielAmah) [inline server/controllers/GroupCtr.js]: * indentation issues
* The file name should match the class name
* catch internal server errors

File: server/config/app.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+
+Routes(router);
+
+// prefix /api for all routes
+app.use('/api', router);
+
+app.get('*', (req, res) => res.status(200).send({
```
Comment (Seunadex) [inline server/config/app.js]: You might want to make this return probably your Api documentation instead of this text

File: server/controllers/GroupCtr.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      }
+    }).then((group) => {
+      if (group) {
+        group.getUsers()
+         .then(groups => res.status(200).send(groups));
+      } else {
+        return res.status(404).send({ message: 'Group does not exist' });
+      }
+    });
+  },
```
Comment (DanielAmah) [inline server/controllers/GroupCtr.js]: * it is obvious you are using the same 404 message errors in most of your methods. Abstract it as a function and call it.
* still indentation

File: server/controllers/MessageCtr.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          }
+        }).then(messages => res.status(200).send(messages));
+      } else {
+        return res.status(404).send({ message: 'Group not found' });
+      }
+    });
+  }
+};
+
+export default MessageCtrl;
```
Comment (DanielAmah) [inline server/controllers/MessageCtr.js]: * the file name should match the class name

File: server/controllers/GroupCtr.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+};
+
+export default GroupCtrl;
```
Comment (Seunadex) [inline server/controllers/GroupCtr.js]: The name what you are exporting is better written in full

File: server/controllers/MessageCtr.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import db from '../models';
+
+
+const MessageCtrl = {
```
Comment (Seunadex) [inline server/controllers/MessageCtr.js]: Ditto, consider renaming this and other files to be in full

File: server/controllers/UserCtr.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  signOut(req, res) {
+    res.status(200).send({
+      message: 'User successfully logged out'
+    });
+  },
+
+};
+
+
+export default UsersCtrl;
```
Comment (DanielAmah) [inline server/controllers/UserCtr.js]: the file name  should match the class name

File: server/controllers/GroupCtr.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+                return res.status(201).send({
+                  message: 'successfully added user to group'
+                });
+              }
+              return res.status(409).send({
+                message: 'user already exist'
+              });
+            });
+          } else {
+            return res.status(400).send({
```
Comment (Seunadex) [inline server/controllers/GroupCtr.js]: Kindly check your status code, make sure it correlate with the error message

File: server/controllers/GroupCtr.js [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+              message: 'User does not exist'
+            });
+          }
+        });
+      } else {
+        return res.status(400).send({
```
Comment (Seunadex) [inline server/controllers/GroupCtr.js]: Ditto, same here

File: server/controllers/MessageCtr.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+          groupId: req.params.id
+        }).then(messageCreated => res.status(201).send({
+          id: messageCreated.id,
+          content: messageCreated.content,
+          priority: messageCreated.priority,
+          isRead: messageCreated.isRead,
+          createdAt: messageCreated.createdAt
+        }));
+      } else {
+        return res.status(409).send({ message: 'Group doesnt exist' });
```
Comment (Seunadex) [inline server/controllers/MessageCtr.js]: Ditto, Kindly review all your status code again

File: server/migrations/20170614074820-create-user.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: queryInterface => queryInterface.dropTable('Users')
+};
```
Comment (DanielAmah) [inline server/migrations/20170614074820-create-user.js]: * your password could have a min length value

File: server/models/index.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+Object.keys(db).forEach((modelName) => {
+  if (db[modelName].associate) {
+    db[modelName].associate(db);
+  }
+});
+
+db.sequelize = sequelize;
+db.Sequelize = Sequelize;
+
+module.exports = db;
```
Comment (DanielAmah) [inline server/models/index.js:47]: * i believe import will be better than required for your modules

File: server/controllers/UserCtr.js [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+        email
+      }
+    }).then((user) => {
+      if (user && helper.validatePassword(user, password)) {
+        res.status(200).send({
+          message: 'You have been loggedin successfully',
+          token: Auth.generateToken(user)
+        });
+      } else {
+        res.status(403).send({
```
Comment (Seunadex) [inline server/controllers/UserCtr.js]: Please, find out when to use status code 403

File: server/tests/server/User.test.js [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+    request.post('/api/v1/user/signup')
+    .send(anotherUser2)
+    .end((err, res) => {
+      if (err) return err;
+      expect(res.body.message).to.equal('Unexpected error occured');
+      return done();
+    });
+  });
+});
+
```
Comment (DanielAmah) [inline server/tests/server/User.test.js:140]: * your test descriptions should be more descriptive to convey what is been tested

File: server/models/message.js [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+      defaultValue: false
+    }
+  });
+  Messages.associate = (models) => {
+    Messages.belongsTo(models.Groups, {
+      foreignKey: 'groupId',
+      onDelete: 'CASCADE'
+    });
+  };
+  return Messages;
```
Comment (Seunadex) [inline server/models/message.js:30]: Please follow Airbnb style guide concerning naming of base file

File: server/models/user.js [hunk 16]
```diff
@@ -0,0 +1,38 @@
+
+module.exports = (sequelize, DataTypes) => {
+  const Users = sequelize.define('Users', {
```
Comment (Seunadex) [inline server/models/user.js:3]: ditto , same here and all your filename of your models

File: server/tests/server/User.test.js [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+  it('should unique username', (done) => {
+    request.post('/api/v1/user/signup')
+    .send(anotherUser2)
+    .end((err, res) => {
+      if (err) return err;
+      expect(res.body.message).to.equal('Unexpected error occured');
+      return done();
+    });
+  });
+});
```
Comment (DanielAmah) [inline server/tests/server/User.test.js]: * ensure that the test description should be more descriptive that what it is

File: server/tests/server/Group.test.js [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+  //           if (err) return err;
+  //           userResponse = res;
+  //           console.log(user);
+  //           done();
+  //         });
+  //       });
+  //   });
+  // });
+});
+
```
Comment (DanielAmah) [inline server/tests/server/Group.test.js]: remove comments from your tests

File: server/tests/Helpers/MessageHelper.js [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+    content: 'i love andela',
+    priority: 'critical',
+  },
+
+  badMessage: {
+    content: 'i love andela',
+    priority: 'critical',
+  },
+
+};
```
Comment (Seunadex) [inline server/tests/Helpers/MessageHelper.js:12]: kindly fix all EOF line error in all your files

File: server/tests/Helpers/UserHelper.js [hunk 20]
```diff
@@ -0,0 +1,37 @@
+import faker from 'faker';
```
Comment (Seunadex) [inline server/tests/Helpers/UserHelper.js]: Kindly fix all indentation error

File: server/tests/Helpers/UserHelper.js [hunk 21]
```diff
@@ -0,0 +1,40 @@
+
+module.exports = {
+  goodUser: {
+    username: 'johnny19',
+	  email: 'chinonso@example.com',
```
Comment (Seunadex) [inline server/tests/Helpers/UserHelper.js]: Kindly fix all indentation error

File: server/tests/server/Group.test.js [hunk 22]
```diff
@@ -0,0 +1,178 @@
...
+      expect(res.body.message).to.equal('Group already exist');
+      return done();
+    });
+  });
+  it('should add users to group', (done) => {
+    request.post('/api/v1/group/1/user')
+    .send(username)
+    .set({ Authorization: userResponse.token })
+    .end((err, res) => {
+      expect(res.status).to.equal(201);
```
Comment (Seunadex) [inline server/tests/server/Group.test.js]: When writing test, it's imperative that what you intend to test for matches what's actually being tested for. It's also important that your test descriptions match what's actually being tested for. Please go through the other tests suites in your code base and refactor appropriately.

File: server/tests/server/Group.test.js [hunk 23]
```diff
@@ -0,0 +1,178 @@
...
+      done();
+    });
+  });
+  it('should not add user to group if user has not signup', (done) => {
```
Comment (Seunadex) [inline server/tests/server/Group.test.js]: Please , let your test description match what you are testing for

File: server/tests/server/Group.test.js [hunk 24]
```diff
@@ -0,0 +1,178 @@
...
+    request.post('/api/v1/group/1/user')
+    .send(userDoesntExist)
+    .set({ Authorization: userResponse.token })
+    .end((err, res) => {
+      expect(res.status).to.equal(409);
+      done();
+    });
+  });
+  it('should not add user to group if user already exist', (done) => {
```
Comment (Seunadex) [inline server/tests/server/Group.test.js]: ditto,

File: server.js [hunk 25]
```diff
@@ -0,0 +1,14 @@
+// This will be our application entry. We'll setup our server here.
```
Comment (andela-dbamidele) [inline server.js:1]: Since this file is doing nothing really special, you can combine it with your `server/config/app.js` file.

File: server.js [hunk 26]
```diff
@@ -0,0 +1,14 @@
...
+import app from './server/config/app';
+
+require('dotenv').config();
+
+const port = process.env.PORT || 5000;
+app.listen(port, (err) => {
+  if (err) {
+    console.log(err);
+  } else {
+    console.log(`App running at http://localhost:${port}`);
```
Comment (andela-dbamidele) [inline server.js]: `console.log` is not suitable for production. Use `Logger` instead

File: server/controllers/GroupCtr.js [hunk 27]
```diff
@@ -0,0 +1,133 @@
...
+
+  /**
+   *  - Create a user
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {void} Returns void
+   */
+  createGroup(req, res) {
+    db.Groups.findOrCreate({
+      where: { name: req.body.name },
```
Comment (andela-dbamidele) [inline server/controllers/GroupCtr.js]: You should validate `name` before querying the database

File: server/controllers/GroupCtr.js [hunk 28]
```diff
@@ -0,0 +1,133 @@
...
+    }));
+  },
+  /** Add User To Group
+   * @param {Object} req Request Object
+   * @param {Object} res Response Object
+   * @returns {object} Returns User
+   */
+
+  addUserToGroup(req, res) {
+    const groupId = req.params.id;
```
Comment (andela-dbamidele) [inline server/controllers/GroupCtr.js]: You should validate `id` to ensure it's a number. Hence sequelize will return error

File: server/controllers/UserCtr.js [hunk 29]
```diff
@@ -0,0 +1,95 @@
...
+
+  /**
+   * showUsers
+   * @desc gets details for all users
+   * @param {Object} req request object
+   * @param {Object} res response object
+   * @returns {void}
+   */
+  showUsers(req, res) {
+    db.Users.findAll({
```
Comment (andela-dbamidele) [inline server/controllers/UserCtr.js]: You can shorten this to `findAll({})`

**Review by andela-dbamidele — COMMENTED** [review]
Nice effort you put in your code. You just need to do a little housekeeping.

Good luck.
